Makefile

```diff
@@ -150,3 +147,16 @@ node_modules/.uptodate: package.json
 	@echo installing javascript dependencies
 	@node_modules/.bin/check-dependencies 2>/dev/null || npm install
 	@touch $@
+
+PYTHON_VERSION = 3.6.8
+PYTHON_MINOR_VERSION := $(shell echo $$PYTHON_VERSION | cut -d '.' -f1,2)
+.PHONY: python
+python:
+	@if [ `python${PYTHON_MINOR_VERSION} --version 2>/dev/null | grep -c "${PYTHON_VERSION}"` -ne 1 ]; then pyenv install -s ${PYTHON_VERSION}; fi
+
+TOX_VERSION := $(shell tox --version 2>/dev/null)
+.PHONY: tox
+tox: python
+ifndef TOX_VERSION
+	pip install tox
+endif
```

Comment:
We use `tox --version` to test whether `tox` is available on $PATH and install it if not. This is simpler than for Python because we don't require a particular version of tox.